### ISSUE TITLE
Duplicate addon styles in vendor.css

### ISSUE THREAD

**Issue by fpauser** [ember-cli/ember-cli#9076]
Addon-styles are appended twice to vendor.css.

Reproduction:
```bash
ember addon dummy-addon --yarn && \
cd dummy-addon && \
mkdir addon/styles && \
echo ".dummy-addon { display: block; }" > addon/styles/dummy-addon.css && \
ember s
```
In another shell:
```
curl http://localhost:4200/assets/vendor.css
```

Output:
```
.dummy-addon { display: block; }

.dummy-addon { display: block; }
```

Output from `ember version --verbose && npm --version && yarn --version`:
```
ember-cli: 3.16.0
node: 12.14.1
v8: 7.7.299.13-node.16
uv: 1.33.1
zlib: 1.2.11
brotli: 1.0.7
ares: 1.15.0
modules: 72
nghttp2: 1.40.0
napi: 5
llhttp: 2.0.1
http_parser: 2.8.0
openssl: 1.1.1d
cldr: 35.1
icu: 64.2
tz: 2019c
unicode: 12.1
os: linux x64
6.13.4
1.22.0
```

**Comment by btecu** [comment]
I've seen this as well using `ember-cli` 3.14.0.

**Comment by chriseppstein** [comment]
I noticed this too.

The bug seems to be because the tree returned by `treeFor('addon')` contains each file in two places:

```
Tree: foo.treeFor("addon")
        foo.css
        foo/styles/addon.css
        foo/styles/other.css
        other.css
```

```
$ cat  lib/foo/addon/styles/other.css
.other {
  border: none;
}

$ cat lib/foo/addon/styles/addon.css
.foo {
  color: red;
}                                                                                                                                                         
$ tail -16 dist/assets/vendor.css

.foo {
  color: red;
}
.foo {
  color: red;
}
.other {
  border: none;
}


.other {
  border: none;
}
```

**Comment by bantic** [comment]
Looked into this with @twokul and we determined that the reason 84c449a introduces this issue is that prior to that change, the `Addon#compileTemplates` method [1] would only contain files in the "templates/*" directory (because it calls `_addonTemplateFiles` [2]). After 84c449a, `compileTemplates` no longer calls `_addonTemplateFiles`, so now it *does* include those `style/**/*` files. When `Addon#treeForAddon` merges addonTree and stylesTree [3], some styles files will be in the output tree multiple times.
When default-package scoops up all the addon CSS files [4] and concats them, the final `vendor.css` file contains the duplicates.

[1] `Addon#compileTemplates` @ 2552a7 (prior to this issue) calls `_addonTemplateFiles ` on the tree: https://github.com/ember-cli/ember-cli/blob/2552a736ef09fed6b5b8ac691f729f516fbbb425/lib/models/addon.js#L1219
[2] `Addon#_addonTemplateFiles` @ current master: [Moves addon templates to `/templates`](https://github.com/ember-cli/ember-cli/blob/3856e4caa29742df8f0dd67c15c6845638b6dac3/lib/models/addon.js#L1087), and [only includes template files from pods](https://github.com/ember-cli/ember-cli/blob/3856e4caa29742df8f0dd67c15c6845638b6dac3/lib/models/addon.js#L1095-L1102), resulting in no `styles/` files in the output
[3] `Addon#treeForAddon` @ current master: https://github.com/ember-cli/ember-cli/blob/3856e4caa29742df8f0dd67c15c6845638b6dac3/lib/models/addon.js#L856-L861
[4] https://github.com/ember-cli/ember-cli/blob/3856e4caa29742df8f0dd67c15c6845638b6dac3/lib/broccoli/default-packager.js#L565

**Comment by rwjblue** [comment]
The fix for this is released in ember-cli@3.16.2 and ember-cli@3.18.0 (thanks again @bantic!).
